src/build.sh

```diff
@@ -247,9 +254,21 @@ function full_cleanup()
   cmd_manager "$flag" "$cmd"
 }
 
+function from_sha()
```

Comment:
Don't forget to add documentation. Also, I think we can find a better name for this function... maybe something like `build_from_sha`